easybuild/easyconfigs/l/LAMMPS/LAMMPS-20180316-foss-2018a-Python-3.6.4.eb

```diff
@@ -0,0 +1,44 @@
+name = 'LAMMPS'
+version = '20180316'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://lammps.sandia.gov/'
+description = """LAMMPS is a classical molecular dynamics code,
+and an acronym for Large-scale Atomic/Molecular Massively Parallel Simulator."""
+
+toolchain = {'name': 'foss', 'version': '2018a'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/lammps/lammps/archive/']
+sources = ['stable_16Mar2018.tar.gz']
+checksums = ['03f91d69f472c37a4008611dee13fad91b6bedc66dbf0393416cdacc751b57ff']
+
+dependencies = [
+    ('Python', '3.6.4'),
+    ('libpng', '1.6.34'),
+    ('libjpeg-turbo', '1.5.3'),
+    ('PCRE', '8.41'),
+    ('libxml2', '2.9.7'),
+    ('zlib', '1.2.11'),
+    ('gzip', '1.8'),
+    ('FFmpeg', '3.4.2'),
+    ('HDF5', '1.10.1'),
+    ('Eigen', '3.3.4', '', True),
+    ('Voro++', '0.4.6'),
+]
+
+# run make in lib/<pkglib> for each of these (and use a particular Makefile, if specified)
+packaged_libraries = ['atc', 'awpmd', 'colvars', 'h5md', 'meam', 'poems', 'qmmm', 'reax']
+
+# run "make yes-<pkg>" for each of these
+packages_yes = ['ASPHERE', 'BODY', 'CLASS2', 'COLLOID', 'COMPRESS', 'CORESHELL', 'DIPOLE', 'GRANULAR',
+                'KSPACE', 'MANYBODY', 'MC', 'MEAM', 'MISC', 'MOLECULE', 'MPIIO', 'PERI', 'POEMS', 'PYTHON',
+                'QEQ', 'REAX', 'REPLICA', 'RIGID', 'SHOCK', 'SNAP', 'SRD', 'USER-ATC', 'USER-AWPMD', 'USER-CG-CMM',
+                'USER-CGDNA', 'USER-COLVARS', 'USER-DIFFRACTION', 'USER-DPD', 'USER-DRUDE', 'USER-EFF', 'USER-FEP',
+                'USER-H5MD', 'USER-LB', 'USER-MANIFOLD', 'USER-MGPT', 'USER-MOLFILE', 'USER-NC-DUMP', 'USER-PHONON',
+                'USER-QMMM', 'USER-QTB', 'USER-REAXC', 'USER-SMD', 'USER-SMTBQ', 'USER-SPH', 'USER-TALLY', 'VORONOI']
+
+# run "make no-<pkg>" for each of these
+packages_no = ['GPU', 'KIM', 'KOKKOS', 'MSCG', 'OPT', 'USER-INTEL', 'USER-MISC', 'USER-OMP', 'USER-QUIP', 'USER-VTK']
```

Comment:
`USER-OMP` provides OpenMP optimized fixes. It may be useful for users running on nodes with 16+ cores.

Reply:
I was going with the packages that the ComputeCanada team was using (cfr. https://github.com/ComputeCanada/easybuild-easyconfigs/blob/computecanada-master/easybuild/easyconfigs/l/LAMMPS).

@mboisson Any idea why `USER-OMP` is not included?

Reply:
Probably some package incompatibility with other packages as well. @alikerr may be able to say more.

Reply:
LAMMPS includes different packages (about 60 for the current version). Some of them can not be included if an external library is not available (like KIM, QUIP, KOKKOS, ...) for the others the libraries should be compiled before LAMMPS under the directory "lib". LAMMPS uses also some accelerators (GPU, USER-OMP, USER-INTEL). Some of them can not be combined with all the available packages. The list of packages change from one version to another. In the very recent stable version, there are at least two new features or packages: Wrapper on the LATTE DFTB and USER-MESO. The easyconfig for LAMMPS and the patch will change with the version however the easyblock lammps.py may not change very often.

Reply:
Most of the time, I use Makefile.icc_openmpi by adding extra parameters (libraries and other flags like).